src/decoder/plugins/SidplayDecoderPlugin.cxx

```diff
@@ -437,51 +439,83 @@ sidplay_file_decode(DecoderClient &client, Path path_fs)
 	} while (cmd != DecoderCommand::STOP);
 }
 
+static AllocatedString<char>
+Windows1252ToUTF8(const char *s)
+{
+#ifdef HAVE_ICU_CONVERTER
+	try {
+		std::unique_ptr<IcuConverter>
+			converter(IcuConverter::Create("windows-1252"));
+
+		return converter->ToUTF8(s);
+	} catch (...) { }
+#endif
+
+	/*
+	 * Fallback to not transcoding windows-1252 to utf-8, that may result
+	 * in invalid utf-8 unless nonprintable characters are replaced.
+	 */
+	AllocatedString<char> t = AllocatedString<char>::Duplicate(s);
+
+	for (size_t i = 0; t[i] != AllocatedString<char>::SENTINEL; i++)
+		if (!std::isprint(t[i]))
+			t[i] = '?';
+
+	return t;
+}
+
 gcc_pure
-static const char *
-GetInfoString(const SidTuneInfo &info, unsigned i) noexcept
+static AllocatedString<char>
+GetInfoString(const SidTuneInfo &info, unsigned i)
 {
 #ifdef HAVE_SIDPLAYFP
-	return info.numberOfInfoStrings() > i
+	const char *s = info.numberOfInfoStrings() > i
 		? info.infoString(i)
-		: nullptr;
```

Comment:
Unlike `std::string`, `AllocatedString<>` is nullable. You don't need to use an empty string as magic value for "no value", but keep `nullptr` here.

Reply:
Yes, but the previous code had a special case to convert the `nullptr` to the empty string, which is why it was simpler to skip the `nullptr` altogether. I consider the empty string less magic (and much safer and less likely to crash anything) than `nullptr`.

And if the retrieved value happens to be the empty string (which can happen), then it does not seem like it should be used anyway, so why not handle the empty string as the common case without introducing the `nullptr` at all?

Reply:
I prefer returning `nullptr` if there is no value, but I'd accept your approach.
(Sure, empty values will be discarded later, but I don't want to allocate memory for an empty string when I already know I won't be using it.)

Reply:
`AllocatedString<>` could be designed to special case the empty string to not allocate anything at all. Then user code could be maximally fast and safe without the `nullptr`. :smile:

Reply:
It could, but that would make its implementation very complicated. Right now, it's a plain pointer, and has nearly no overhead.